grid2op/Opponent/OpponentSpace.py

```diff
@@ -107,14 +119,35 @@ def attack(self, observation, agent_action, env_action):
             The attack the opponent wants to perform (or "do nothing" if the attack was too costly)
 
         """
+
+        # Update variables
         self.budget += self.budget_per_timestep
-        attack = self.opponent.attack(observation, agent_action, env_action, self.budget, self.previous_fails)
-        tmp = self.compute_budget(attack)
-        if tmp <= self.budget:
-            self.previous_fails = False
-            self.budget -= tmp
-            res = attack
+        self.current_attack_duration = max(0, self.current_attack_duration - 1)
+        self.current_attack_cooldown = max(0, self.current_attack_cooldown - 1)
+
+        # If currently attacking
+        if self.current_attack_duration > 0:
+            attack = self.current_attack
+
+        # If the opponent has already attacked today
+        elif self.current_attack_cooldown > self.attack_cooldown:
+            attack = self._do_nothing
+
+        # If the opponent can attack  
         else:
-            self.previous_fails = True
-            res = self._do_nothing
-        return res
+            self.previous_fails = False
+            attack = self.opponent.attack(observation, agent_action, env_action, self.budget,
+                                          self.previous_fails)
+            # If the cost is too high
+            if self.attack_duration * self.compute_budget(attack) > self.budget:
+                attack = self._do_nothing
+                self.previous_fails = True
+            # If we can afford the attack
+            elif attack != self._do_nothing:
+                self.current_attack_duration = self.attack_duration
+                self.current_attack_cooldown += self.attack_cooldown
```

Comment:
change this part to say "if not attack then `None`"
It will clarify the code and make it run faster.

Reply:
Done